### ISSUE TITLE
Array properties with array items listed in enums does not validate properly

### ISSUE THREAD

**Issue by malinink** [justinrainbow/json-schema#547]
For example if we have array property:
```
{"data": ["a", "b"]}
```
And validation like:
```
{
    "type": "object",
    "properties": {
        "data": {
            "type": "array",
            "items": {
                "type": "string",
                "enum": ["b", "c"]
            }
        }
    }
}
```

Validation wrongly finished with success.
I will provide PR with test for that particular case.

**Comment by erayd** [comment]
Thanks - good catch! That should definitely not be passing validation.

**Comment by erayd** [comment]
Fixed in master with #498, backported for 5.2.8 in #558.

**Comment by rbchasesc** [comment]
@erayd Finally got to writing some test cases for these. Can be found in #559

**Comment by malinink** [comment]
@erayd  tests are OK, there is no need of that Issue and PR.

**Comment by erayd** [comment]
@bighappyface Could you please close #547 (fixed in #498) and #549 (superseded by #559), and merge #559.
